pysmt/solvers/btor.py

```diff
@@ -0,0 +1,347 @@
+            if name_filter is None or not var.symbol_name().startswith(name_filter):
+                print("%s = %s" % (var.symbol_name(), self.get_value(var)))
+
+    def get_value(self, item):
+        self._assert_no_function_type(item)
+        return self.converter.back(item)
+
+    def exit(self):
+        if not self._destroyed:
+            self._destroyed = True
+            del self.btor
+
+
+class BTORConverter(Converter, DagWalker):
+
+    def __init__(self, environment, btor):
+        DagWalker.__init__(self, environment)
+        self.mgr = environment.formula_manager
+        self._get_type = environment.stc.get_type
+        self._back_memoization = {}
+        self._btor = btor
+        self.declared_vars = {}
+        return
+
+    @catch_conversion_error
+    def convert(self, formula):
+        return self.walk(formula)
+
+    def back(self, expr):
+        titem = self.convert(expr)
+        if expr.is_symbol() and  expr.symbol_type().is_bv_type():
+            return self.mgr.BV(titem.assignment, expr.bv_width())
+        else:
+            return self.mgr.Bool(bool(int(titem.assignment)))
+
+    def walk_and(self, formula, args, **kwargs):
+        #pylint: disable=star-args
+        assert len(args) >= 2
+        res = self._btor.And(args[0], args[1])
+        for conj in args[2:]:
+            res = self._btor.And(res, conj)
+        return res
+
+    def walk_or(self, formula, args, **kwargs):
+        #pylint: disable=star-args
+        assert len(args) >= 2
+        res = self._btor.Or(args[0], args[1])
+        for disj in args[2:]:
+            res = self._btor.Or(res, disj)
+        return res
+
+    def walk_not(self, formula, args, **kwargs):
+        return self._btor.Not(args[0])
+
+    def walk_symbol(self, formula, **kwargs):
+        symbol_type = formula.symbol_type()
+        if symbol_type.is_bool_type():
+            res = self._btor.Var(1, formula.symbol_name())
+        elif symbol_type.is_real_type():
+            raise ConvertExpressionError
+        elif symbol_type.is_int_type():
+            raise ConvertExpressionError
+        else:
+            assert symbol_type.is_bv_type()
+            res = self._btor.Var(formula.bv_width(),
+                                 formula.symbol_name())
+        self.declared_vars[formula] = res
+        return res
+
+    def walk_iff(self, formula, args, **kwargs):
+        return self._btor.Iff(*args)
+
+    def walk_implies(self, formula, args, **kwargs):
+        return self._btor.Implies(*args)
+
+    def walk_ite(self, formula, args, **kwargs):
+        return self._btor.Cond(*args)
+
+    def walk_bool_constant(self, formula, **kwargs):
+        return self._btor.Const(formula.constant_value())
+
+    def walk_equals(self, formula, args, **kwargs):
+        return self._btor.Eq(*args)
+
+    def walk_function(self, formula, args, **kwargs):
+        #pylint: disable=star-args
+        raise NotImplementedError
+        f = formula.function_name()
```

Comment:
This is indeed wrong. Boolector supports functions. I'll extend the logic to QF_UFBV and implement this missing piece.